packages/server/src/tests/utilities/TestConfiguration.ts

```diff
@@ -428,6 +428,34 @@ export default class TestConfiguration {
 
   // HEADERS
 
+  // sets the role for the headers, for the period of a callback
+  async loginAsRole(roleId: string, cb: () => Promise<unknown>) {
+    const roleUser = await this.createUser({
+      roles: {
+        [this.prodAppId!]: roleId,
```

Comment:
`prodAppId` is often undefined because I believe it's only set when `config.publish` is called. I'd call `getProdAppId()` so there's proper error checking, or `getAppId()` if it's not super important to be the prod one.